apps/wallet/src/ui/app/hooks/useSigner.ts

```diff
@@ -1,12 +1,17 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useActiveAddress } from './useActiveAddress';
 import { thunkExtras } from '_redux/store/thunk-extras';
 
-export function useSigner() {
-    const { api, keypairVault, background } = thunkExtras;
-    return api.getSignerInstance(
-        keypairVault.getKeypair().getPublicKey().toSuiAddress(),
-        background
-    );
+import type { SuiAddress } from '@mysten/sui.js';
+
+export function useSigner(address?: SuiAddress) {
+    const activeAddress = useActiveAddress();
+    const signerAddress = address || activeAddress;
+    const { api, background } = thunkExtras;
+    if (!signerAddress) {
```

Comment:
What is the case where this isn't present?

Reply:
When wallet is locked or uninitialised